lib/ansible/modules/windows/win_domain_computer.ps1

```diff
@@ -0,0 +1,183 @@
+#!powershell
+#
+# Copyright: (c) 2017, AMTEGA - Xunta de Galicia
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+#Requires -Module Ansible.ModuleUtils.Legacy
+
+
+# ------------------------------------------------------------------------------
+$ErrorActionPreference = "Stop"
+
+# Preparing result
+$result = @{}
+Set-Attr $result "changed" $false
+
+# Parameter ingestion
+$params = Parse-Args $args -supports_check_mode $true
+
+$check_mode = Get-AnsibleParam -obj $params -name "_ansible_check_mode" -default $false | ConvertTo-Bool
+$diff_support = Get-AnsibleParam -obj $params -name "_ansible_diff" -default $false | ConvertTo-Bool
+
+# ------------------------------------------------------------------------------
+Function Get-DesiredState($params) {
+  $name = Get-AnsibleParam -obj $params -name "name" -failifempty $true -resultobj $result
```

Comment:
One things that makes this module difficult for people who would need to maintain it is that the options are in a different location to where they are normally put for Windows modules. Usually we define the options straight after `$params = Parse-Args $args -supports_check_mode $true` and then pass them into function from there. Doing this has the benefits of

* Making it clear for anyone looking at the code to see what options are in use
* Easily reconcile the actual modules and the ones in the docs as well as the default/choices for these options
* Keeps things consistent with the other Windows modules, making it easier to maintain in the future

Reply:
Like it less because it's less encapsulation, but `Readability counts`.